cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/PipelineConfigurer.java

```diff
@@ -31,4 +33,18 @@
    * @return stage configurer
    */
   StageConfigurer getStageConfigurer();
+
+  /**
+   * @return the engine for this pipeline
+   */
+  Engine getEngine();
+
+  /**
+   * Set pipeline properties that will be applied to each run of the pipeline.
+   * Depending on the engine used, the properties will be added to the SparkConf for each run or the mapreduce
```

Comment:
are all properties added to the config or only once which are prefixed `system.spark`/`system.mapreduce`?